### ISSUE TITLE
Some Religion UI Follower windows missing truncated text

### ISSUE THREAD

**Issue by deadstarre** [LoneGazebo/Community-Patch-DLL#4171]
_1. Mod version (i.e Date - 4/23):_
4/5
_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description:_ 

Most of the follower windows will automatically truncate any excess text into "..." but i found if you add lines of text to a few of them they will not truncate but display all text in its entirety, which spills over into the next windows and looks pretty ugly.  

Not an issue in the current VP setup (affected windows have short desciptions in VP) but I'm hoping its a very easy thing to do

the windows I found that refuse to truncate are:  Thrift, Cooperation and Asceticism, but ideally they should all be able to truncate. 

_4. Steps to reproduce (optional):_
![respect](https://user-images.githubusercontent.com/15820785/38633123-b1f65242-3d8c-11e8-948f-aec00acbbb72.jpg)

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Comment by LoneGazebo** [comment]
So all followers? It’s probably a LUA setting that causes it for follower beliefs in the ReligionOverview.xml file.

**Comment by deadstarre** [comment]
it's just that most of them truncate for that window but a few aren't, probably some setting vanilla never bothered initializing for the windows it wasn't needed in. This is basically a feature request, I just checked ReligionOverview and im not seeing anything obvious for it there. Then again I dont really understand most of what i'm looking at... perhaps @Infixo could be enticed to take a look?

**Comment by LoneGazebo** [comment]
In ReligionOverview.xml, find this line:

`<Label Anchor="L,C" Font="TwCenMT18" TruncateWidth="425" ColorSet="Beige_Black_Alpha" FontStyle="Shadow" ID="BeliefDescription" />`

make the TruncateWidth match the sizes of the other two above it (so around 150. See if that fixes it for you.

**Comment by deadstarre** [comment]
I was looking at ReligionOverview.lua before... 

just to be clear, youre talking about the vanilla file? if VP is overwriting that anywhere i cannot find it, i checked EUI also and see nothing, I only see the ReligionOverview.lua. 

I did try editing the vanilla ReligionOverview.xml file at Assets\DLC\Expansion2\UI\InGame\Popups where you suggested to 150 but unfortunately no effect. i also tried lowering wrapwidth down from 450 (which altered my personal religion overview, so i guess editing this file does do something) and i tried lowering truncate in a different spot as well and also saw nothing.

its only an issue for a belief once i've selected it, like in my picture above. the belief tooltips on the right hand side (missing from my picture above) handle any extra text perfectly

**Comment by deadstarre** [comment]
ah, also it isn't all the followers. Veneration for example I can add rows of text to and it still truncates once selected

**Comment by Infixo** [comment]
I actually don’t understand what the problem is? TextBox truncates the text if it’s too long. From .xml it seems like 425, so around half-width of the window. And it works, as on the screenshot (3 dots). @deadstarre Can you describe how in your opinion it should look like?

**Comment by deadstarre** [comment]
here we go. notice that every belief on the screen correctly truncates, and yet Thrift doesn't- the extra text just carries right into the following box, no 3 dots. 

I've only noticed this for Asceticism, Thrift and Cooperation beliefs.

![respect](https://user-images.githubusercontent.com/15820785/38647139-d34ff4da-3db8-11e8-975b-c117d986c161.jpg)

**Comment by Infixo** [comment]
The window in question is actually ChooseReligionPopup.xml. VP doesn't modify the .xml. Vp modifies a bit .lua, but only to add tooltips, rest of the code is the same as Expansion2 version.
Also, each box with beliefs is basically the same:

```
<Button Color="0,0,0,255" Size="500,50" ID="FollowerBelief2Button">
<!-- cut -->
<Label Anchor="L,T" Offset="24,7" TruncateWidth="460" Font="TwCenMT20" ColorSet="Beige_Black_Alpha" FontStyle="Shadow" ID="FollowerBelief2Name" String="TXT_KEY_CHOOSE_RELIGION_FOLLOWER_BELIEF2"/>
<Label Anchor="L,T" Offset="52,28" TruncateWidth="460" Font="TwCenMT16" ColorSet="Beige_Black_Alpha" FontStyle="Shadow" ID="FollowerBelief2Description" String="TXT_KEY_CHOOSE_RELIGION_SELECT_FOLLOWER_BELIEF2"/>
</Button>
```
I've checked and every one has 2 Labels, each with Truncate set to 460.

**Comment by LoneGazebo** [comment]
Ah, I thought he was in the religionoverview.

**Comment by deadstarre** [comment]
I appreciate you taking a look. no mods or manual changes, i think we can rule out that option, you could verify it with your install by editing the ReligionText.sql in VP, find this entry and add anything you'd like into the middle of the description, look ingame after selecting Asceticism and any extra text will be overflowed like my pic

UPDATE Language_en_US
SET Text = '+1 [ICON_FOOD] Food for every follower in the City (max +10 [ICON_FOOD] Food)'
WHERE Tag = 'TXT_KEY_BELIEF_ASCETISM'

about the code you just posted, it ends with TXT_KEY_CHOOSE_RELIGION_SELECT_FOLLOWER_BELIEF2 which is "Select a Follower Belief." so yes thats the correct box we're tallking about, but it must change after I select a belief, because the box goes from reading "Select a Follower Belief" to importing & displaying the exact description of which belief i've selected. 

and since some follower beliefs truncate and some don't, I have to think its during that import process where some settings change and the problem occurs.

**Comment by deadstarre** [comment]
@Infixo 

yes, using [NEWLINE]. I got thrown for a loop because I used [NEWLINE] in all of my descriptions, and yet some of them would still truncate just fine and some of them weren't. 

Now that you say this to me, I can see the problem exactly: If I use [NEWLINE] -before- the description hits the length at which it would normally truncate, a ton of my text spills over. But if I insert it -after- it would have been truncated already, no problems whatsoever. I think armed with this information now, I can find a way to lengthen the texts a few characters so they reach truncate limit, and -then- insert my [NEWLINE]'s =) 

thank you again!
